Feat/update

tvmalmeida opened the pull request:
PR to adjust registers that check if restart is required

Commit:
.yamlint file defined

Commit:
Pipfiles defined

Commit:
README file defined

Commit:
default variables defined

Commit:
handlers tasks defined

Commit:
main task file defined

Commit:
tests directory deleted

Commit:
vars files defined

Commit:
metadata main file defined

gaoneto commented on tasks/main.yml:
```diff
@@ -1,28 +1,45 @@
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootRequired") `
+      -or (Test-Path "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\WindowsUpdate\Auto Update\RebootPending")
+      if ($RebootRequired) {
+          Write-Output "YES"
```
I suggest using the `$Ansible.Result` variable that is available using the `win_powershell` module instead of a `Write-Output`command. Also avoid using an `else` statement for these cases. Example code for this:

```
$Ansible.Result = $false
if ($RebootRequired) {
  $Ansible.Result = $true
}
```

tvmalmeida replied:
Code changed as suggested.

gaoneto commented on tasks/main.yml:
```diff
@@ -1,28 +1,45 @@
+- name: Reboot host before check for update
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
+  when: reboot_needed.output is defined and "YES" in reboot_needed.output
```
By using the `$Ansible.Result` variable the only thing you need here is:

```
when: reboot_needed.result
```

tvmalmeida replied:
Code changed as suggested.

gaoneto commented on tasks/main.yml:
```diff
@@ -1,28 +1,45 @@
-  when: ansible_reboot_pending
+- name: Reboot host before check for update
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
```
Is this really long timeout necessary? Waiting 1 hour seems really odd and something should be investigated. Using the default value (600) seems really reasonable, as per the docs: This timeout is evaluated separately for both reboot verification and test command success so maximum clock time is actually twice this value. ([ref](https://docs.ansible.com/ansible/latest/collections/ansible/windows/win_reboot_module.html#parameter-reboot_timeout))

gaoneto replied:
I understand that for specific cases a machine may be slow and have lots of updates to apply, hence you may need to wait a long time. Possibly ignore this comment then

tvmalmeida replied:
Considering that some reboots take a long time in the update process, I consider it more prudent to change to 30 min. The interesting thing about this module is that if the reboot takes only 10 min, it will not wait 30 min, but only 10 min.

gaoneto commented on tasks/main.yml:
```diff
@@ -1,28 +1,45 @@
-  win_reboot:
-    reboot_timeout_sec: 1500
+  ansible.windows.win_reboot:
+    reboot_timeout_sec: 3600
```
Ditto

tvmalmeida replied:
Considering that some reboots take a long time in the update process, I consider it more prudent to change to 30 min. The interesting side about this module is that if the reboot takes only 10 min, it will not wait 30 min, but only 10 min.

gaoneto reviewed (commented):
Added some suggestions and doubts to be evaluated

tvmalmeida committed:
reboot_timeout variable and return of script powershell changed

Reboot_timeout variable changed to 1800 and check if reboot is required changed

gaoneto reviewed (approved):
LGTM
